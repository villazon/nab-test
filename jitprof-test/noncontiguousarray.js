var a = [];

a[100] = 1;

var b = [1,2,3];

b.push(1);
