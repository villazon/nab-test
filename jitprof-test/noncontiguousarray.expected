---------------------------
Report of making non-contiguous array:
 * [location: (jitprof-test/noncontiguousarray.js:3:1:3:7)] <- No. usages: 1
...
Number of putting non-contiguous array statements: 1
[****]NonContArray: 1
Why: In order to handle large and sparse arrays, there are two types of array storage internally:
	 * Fast Elements: linear storage for compact key sets
	 * Dictionary Elements: hash table storage otherwise
It's best not to cause the array storage to flip from one type to another.
